dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientBlockingChannel.java

```diff
@@ -115,6 +118,17 @@ public Response execute(Endpoint endpoint, Request request) throws IOException {
             // not retried by default, so ours implements SafeLoggable and retains the simple-name for
             // cleaner metrics.
             throw new SafeConnectTimeoutException(e, failureDiagnosticArgs(endpoint, request, startTime));
+        } catch (NoHttpResponseException e) {
```

Comment:
Are there any other exceptions which we should check for specifically? I did not realize that we retry most exceptions for non-idempotent requests.